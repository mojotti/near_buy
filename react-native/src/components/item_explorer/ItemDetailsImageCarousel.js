import React from 'react';
import { Dimensions } from 'react-native';
import PropTypes from 'prop-types';
import Carousel from 'react-native-looped-carousel';
import ImagePlaceholder from '../image_placeholder/ImagePlaceholder';
import { styles } from '../../static/styles/ItemDetailsImageCarouselStyles';

const { width } = Dimensions.get('window');


export default class ItemDetailsImageCarousel extends React.Component {
  render() {
    return (
      <Carousel
        style={styles.size}
        autoplay={false}
        pageInfo
        currentPage={0}
        pageInfoBackgroundColor={'#292929'}
        pageInfoTextStyle={{ color: '#FFFFFF' }}
      >
        {this.props.images.map((uri, i) => (
          <ImagePlaceholder
            url={uri}
            styles={styles.size}
            placeholderSize={width}
            key={i}
          />
        ))}
      </Carousel>
    );
  }
}

ItemDetailsImageCarousel.propTypes = {
  images: PropTypes.arrayOf(PropTypes.string).isRequired,
};
