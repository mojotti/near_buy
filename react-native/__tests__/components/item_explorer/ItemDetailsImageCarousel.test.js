import React from 'react';
import { shallow } from 'enzyme';
import ItemDetailsImageCarousel
  from '../../../src/components/item_explorer/ItemDetailsImageCarousel';


describe('<ItemDetailsImageCarousel />', () => {
  test('renders carousel when it should', () => {
    const wrapper =
      shallow(<ItemDetailsImageCarousel images={['foo.png', 'bar.png']} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('renders image when it should', () => {
    const wrapper =
      shallow(<ItemDetailsImageCarousel images={['foo.png']} />);

    expect(wrapper).toMatchSnapshot();
  });
});
