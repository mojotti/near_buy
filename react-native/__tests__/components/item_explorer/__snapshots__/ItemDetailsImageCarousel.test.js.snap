// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemDetailsImageCarousel /> renders carousel when it should 1`] = `
<Carousel
  arrows={false}
  autoplay={false}
  bullets={false}
  currentPage={0}
  delay={4000}
  isLooped={true}
  leftArrowText=""
  pageInfo={true}
  pageInfoBackgroundColor="#292929"
  pageInfoTextSeparator=" / "
  pageInfoTextStyle={
    Object {
      "color": "#FFFFFF",
    }
  }
  rightArrowText=""
  style={
    Object {
      "height": 750,
      "width": 750,
    }
  }
  swipe={true}
>
  <ImagePlaceholder
    key="0"
    placeholderSize={750}
    styles={
      Object {
        "height": 750,
        "width": 750,
      }
    }
    url="foo.png"
  />
  <ImagePlaceholder
    key="1"
    placeholderSize={750}
    styles={
      Object {
        "height": 750,
        "width": 750,
      }
    }
    url="bar.png"
  />
</Carousel>
`;

exports[`<ItemDetailsImageCarousel /> renders image when it should 1`] = `
<View
  style={
    Object {
      "height": 750,
      "width": 750,
    }
  }
>
  <ImagePlaceholder
    placeholderSize={750}
    styles={
      Object {
        "height": 750,
        "width": 750,
      }
    }
    url="foo.png"
  />
</View>
`;
